docs/howto/upgrade_from_0_8_to_0_10.md

```diff
@@ -0,0 +1,248 @@
+[Back to Guides](../README.md)
+
+# How to migrate from `0.8` to `0.10` safely
+
+## Disclaimer
+### Proceed at your own risk
+This document attempts to outline steps to upgrade your app based on the collective experience of
+developers who have done this already. It may not cover all edge cases and situation that may cause issues, 
+so please proceed with a certain level of caution. 
+
+## Overview
+This document outlines the steps needed to migrate from `0.8` to `0.10`. The method described
+below has been created via the collective knowledge of contributions of those who have done
+the migration successfully. The method has been tested specifically for migrating from `0.8.3`
+to `0.10.2`.
+
+The high level approach is to upgrade to `0.10` and change all serializers to use 
+a backwards-compatible `VersionEightSerializer`or `VersionEightCollectionSerializer` 
+and a `VersionEightAdapter`. After a few more manual changes, you should have the same
+functionality as you had with `AMS 0.8`. Then, you can continue to develop in your app by creating
+new serializers that don't use these backwards compatible versions and slowly migrate
+existing serializers to the `0.10` versions as needed.
+
+### `0.10` breaking changes
+- Passing a serializer to `render json:` is no longer supported
+    - Ex. `render json: CustomerSerializer.new(customer)`
+- Passing a nil resource to serializer now fails
+    - Ex. `CustomerSerializer.new(nil)`
+- Attribute methods are no longer accessible from other serializer methods
+    - Ex. 
+    ```ruby
+    class MySerializer
+      attributes :foo, :bar
+      
+      def foo
+        bar + 1
+      end
+    end
+    ```
+ - `root` option to collection serializer behaves differently
+    - Ex. `ActiveModel::ArraySerializer.new(resources, root: "resources")`
```

Comment:
My intention with these examples was to show situations that might have existing with `0.8` but would break in `0.10`. I believe in `0.8`, it was still `ActiveModel::ArraySerializer`